plutus-use-cases/src/Language/PlutusTx/Coordination/Contracts/MultiSig.hs

```diff
@@ -0,0 +1,160 @@
+{-# LANGUAGE FlexibleContexts    #-}
+{-# LANGUAGE DataKinds           #-}
+{-# LANGUAGE DeriveGeneric       #-}
+{-# LANGUAGE OverloadedStrings   #-}
+{-# LANGUAGE RecordWildCards     #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TemplateHaskell     #-}
+-- | Implements an n-out-of-m multisig contract.
+module Language.PlutusTx.Coordination.Contracts.MultiSig
+    ( MultiSig(..)
+    , msValidator
+    , msDataScript
+    , msRedeemer
+    , msAddress
+    , lock
+    , initialise
+    , unlockTx
+    -- * Emulator traces
+    , threeOutOfFive
+    ) where
+
+import           Control.Lens
+import           Control.Monad                (void)
+import qualified Data.Map                     as Map
+import qualified Data.Set                     as Set
+import           Data.Foldable                (fold)
+import qualified Ledger.Ada                   as Ada
+import qualified Ledger.Crypto                as Crypto
+import qualified Ledger.Value                 as Value
+import qualified Language.PlutusTx            as P
+import           Ledger                       as Ledger hiding (initialise, to)
+import           Ledger.Validation            as V
+import           Wallet.API                   as WAPI
+import qualified Wallet.Emulator.Types        as EM
+
+data MultiSig = MultiSig 
+                { signatories :: [Ledger.PubKey]
+                -- ^ List of public keys of people who may sign the transaction
+                , requiredSignatures :: Int
+                -- ^ Minimum number of signatures required to unlock
+                --   the output (should not exceed @length signatories@)
+                }
+P.makeLift ''MultiSig
+
+msValidator :: MultiSig -> ValidatorScript
+msValidator sig = 
+  ValidatorScript (Ledger.applyScript mkValidator (Ledger.lifted sig)) where
+    mkValidator = Ledger.fromCompiledCode ($$(P.compile [||
+      let 
+        validate :: MultiSig -> () -> () -> PendingTx -> ()
+        validate (MultiSig keys num) () () p = 
+            let 
+                present = $$(P.length) ($$(P.filter) ($$(V.txSignedBy) p) keys)
+            in
+                if $$(P.geq) present num
+                then ()
+                else $$(P.error) ($$(P.traceH) "WRONG!" ())
+
+      in
+        validate
+      ||]))
+
+-- | Multisig data script (unit value).
+msDataScript :: DataScript
+msDataScript = DataScript $ Ledger.lifted ()
+
+-- | Multisig redeemer (unit value).
+msRedeemer :: RedeemerScript
+msRedeemer = RedeemerScript $ Ledger.lifted ()
+
+-- | The address of a 'MultiSig' contract.
+msAddress :: MultiSig -> Address
+msAddress = Ledger.scriptAddress . msValidator
+
+-- | Lock some funds in a 'MultiSig' contract.
+lock :: (WalletAPI m, WalletDiagnostics m) => MultiSig -> Value -> m ()
+lock ms vl = payToScript_ defaultSlotRange (msAddress ms) vl msDataScript
+
+-- | Instruct the wallet to start watching the contract address
+initialise :: (WalletAPI m) => MultiSig -> m ()

```

Comment:
This isn't needed, right?

Reply:
It is now that I've changed `unlockTx` to run in `WalletAPI`.